doc/sphinx/language/gallina-extensions.rst

```diff
@@ -869,255 +870,297 @@ Module system
+   in a separate :cmd:`Include` command.
 
-       Defines a functor with parameters given by the list of :token:`module_binding` (possibly none), and output module type :token:`module_type`,
-       with body :token:`module_expression`.
+.. cmd:: Include Type @module_type_inl {* <+ @module_type_inl }
 
-    .. cmdv:: Module @ident {* @module_binding} <: {+<: @module_type} := @module_expression
+   .. deprecated:: 8.3
 
-       Defines a functor with parameters given by module_bindings (possibly none) with body :token:`module_expression`.
-       The body is checked against each :n:`@module_type__i`.
+      Use :cmd:`Include` instead.
 
-    .. cmdv:: Module @ident {* @module_binding} := {+<+ @module_expression}
+.. cmd:: Declare Module {? {| Import | Export } } @ident {* @module_binder } : @module_type_inl
 
-       is equivalent to an interactive module where each :token:`module_expression` is included.
+   Declares a module :token:`ident` of type :token:`module_type_inl`.
 
-.. cmd:: Module Type @ident
+   If :n:`@module_binder`\s are specified, declares a functor with parameters given by the list of
+   :token:`module_binder`\s.
 
-   This command is used to start an interactive module type :token:`ident`.
+.. cmd:: Import {+ @qualid }
 
-   .. cmdv:: Module Type @ident {* @module_binding}
+   If :token:`qualid` denotes a valid basic module (i.e. its module type is a
+   signature), makes its components available by their short names.
 
-      Starts an interactive functor type with parameters given by :token:`module_bindings`.
+   .. example::
 
+      .. coqtop:: reset in
 
-Reserved commands inside an interactive module type:
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+         Module Mod.
+         Definition T:=nat.
+         Check T.
+         End Mod.
+         Check Mod.T.
 
-.. cmd:: Include @module
+      .. coqtop:: all
 
-   Same as ``Include`` inside a module.
+         Fail Check T.
+         Import Mod.
+         Check T.
 
-.. cmd:: Include {+<+ @module}
+   Some features defined in modules are activated only when a module is
+   imported. This is for instance the case of notations (see :ref:`Notations`).
 
-   This is a shortcut for the command :n:`Include @module` for each :token:`module`.
+   Declarations made with the :attr:`local` attribute are never imported by the :cmd:`Import`
+   command. Such declarations are only accessible through their fully
+   qualified name.
 
-.. cmd:: @assumption_token Inline @assums
-   :name: Inline
+   .. example::
 
-   The instance of this assumption will be automatically expanded at functor application, except when
-   this functor application is prefixed by a ``!`` annotation.
+      .. coqtop:: in
 
-.. cmd:: End @ident
+         Module A.
+         Module B.
+         Local Definition T := nat.
+         End B.
+         End A.
+         Import A.
 
-   This command closes the interactive module type :token:`ident`.
+   .. coqtop:: all fail
```

Comment:
This `coqtop` block is not indented enough and appears out of the example block even though it is fully relevant to the example.

Reply:
Indented but still doesn't look right.